### PR TITLE
Add tcl highlighting

### PR DESCRIPTION
**Pull request by Eloitor**

### CODE REVIEW

**Comment by Eloitor** [comment]
Is it possible to detect the following line in the second line of a script?
```
# -*-Mode: TCL;-*-
```
Some files created with Emacs have it...

**Comment by the-mikedavis** [comment]
Not currently, that's roughly covered by https://github.com/helix-editor/helix/issues/729

File: runtime/queries/tcl/highlights.scm
Comment (the-mikedavis) [inline runtime/queries/tcl/highlights.scm:1]: The captures used here need to be adapted to the ones we use in Helix: https://docs.helix-editor.com/master/themes.html#syntax-highlighting

Also the `@spell` capture can be removed

File: runtime/queries/tcl/indents.scm
Comment (the-mikedavis) [inline runtime/queries/tcl/indents.scm:1]: We use a different indent system than neovim: https://docs.helix-editor.com/master/guides/indent.html

So these queries will need to re rewritten or removed (if it doesn't exist then the default indent system will kick in)

File: runtime/queries/tcl/highlights.scm
Comment (Eloitor) [reply runtime/queries/tcl/highlights.scm:1]: Did I resolved this?

File: runtime/queries/tcl/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+(escaped_character) @constant.character.escape
+
+[
+ "{" "}"
+ "[" "]"
+ ";"
+ ] @punctuation.delimiter
+
+((simple_word) @constant.numberic
+               (#lua-match? @constant.numberic "^[0-9]+$"))
```
Comment (the-mikedavis) [inline runtime/queries/tcl/highlights.scm:140]:
```suggestion
               (#match? @constant.numberic "^[0-9]+$"))
```

`#lua-match?` is specific to neovim but I believe this regex is already rust-regex compatible. We can just switch to the `#match?` predicate

File: runtime/queries/tcl/indents.scm [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ (command)
+ (conditional)
+ (foreach)
+ (while)
+ (try)
+ (procedure)
+ (command_substitution)
+ ] @indent
+
+[ "}" "]" ] @outindent
```
Comment (the-mikedavis) [inline runtime/queries/tcl/indents.scm:13]:
```suggestion
[ "}" "]" ] @outdent
```

I believe this is just a typo

**Review by the-mikedavis — COMMENTED** [review]
Yep the new queries look good. I just have a few minor comments

**Review by the-mikedavis — DISMISSED** [review]

File: runtime/queries/tcl/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+ "{" "}"
+ "[" "]"
+ ";"
+ ] @punctuation.delimiter
+
+((simple_word) @constant.numberic
+               (#match? @constant.numberic "^[0-9]+$"))
+
+((simple_word) @contant.builtin.boolean
+               (#any-of? @boolean.builtin.boolean "true" "false"))
```
Comment (pascalkuthe) [inline runtime/queries/tcl/highlights.scm:143]:
```suggestion
((simple_word) @contant.builtin.boolean
               (#any-of? @constant.builtin.boolean "true" "false"))
```

**Review by pascalkuthe — COMMENTED** [review]
needs a rebase and still has some problems with the queries

**Comment by Eloitor** [comment]
Thanks, I did the rebase. I don't know how to solve the problems...

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]

**Pull request merged**